### DISCUSSION TITLE
Slider head movement doesn't work well on some (already ranked) beatmaps

### DISCUSSION THREAD

**Discussion by tovc** [ppy/osu#23320] [Ideas] [answered]
osu!lazer changes slider heads such that they can be clicked while moving. While this may be 'proper' functionality, it cuts into the back end of the hit window for slider heads. On slow sliders, this is mostly ( but not entirely ) a non issue. On very fast sliders, such as those found in higher difficulty 'tech' maps, this can result in the back end of a slider head's hit window being nearly entirely cut off. This issue parallels the infamous issue with osu!stable's short sliders, which players usually played by deliberately hitting early. There was solace in that you wouldn't lose acc by playing them that way. In this case, losing acc is the only practical way to play such patterns.

![image](https://user-images.githubusercontent.com/49025619/234688035-f70b38e7-7ee7-4bc4-9bfb-d9bd5ed95ff1.png)
the hit window of a slider head at OD8 should be ±119.5ms. In this example, the hit window is effectively -119.5ms to +32ms, as the slider head has entirely left the area where the head was. Not only does this result in a hit window nearly 40% smaller than that is intended, it also results in a hugely asymmetrical hit window.

the now infamous fast sliders in [t+pazolite - Oshama Scramble ! (IOException Edit) [Special] mapped by IOException](https://osu.ppy.sh/beatmapsets/1376308#osu/2844649) are an example of a pattern that is largely impractical to play by hitting on time. Even if you want to cry 'skill issue', they can still result in a miss for something that _feels_ like it should have at least been a 100.

I am aware that the classic mod's 'no slider head movement' functionality fixes this, but I believe that it should be default functionality, not that hidden away in a pile of mods that many players won't even touch.

**Comment by peppy** [comment]
We are aware of this and tracking it as a gameplay difference: https://github.com/ppy/osu/wiki/Gameplay-differences-from-osu!stable#slider-head-circle-follows-track-snaking-when-not-hit

But I'm having trouble understanding "hidden away in a pile of mods". Are you saying that classic mod is hard to find, or am I misunderstanding something?

The idea of classic mod is that it can be applied to edge case maps like this to fix issues like this.

**Reply by tovc** [reply]
'hidden away' in that it's not something that a lot of newer players are going to find easily. I have a friend who has started playing osu! recently, and he exclusively plays lazer. One of the complaints I've heard is that the mods section is overwhelming. For someone like you or I, the classic mod is a no-brainer, but it's not so clear for someone that hasn't played stable for years.

Beyond that, this really isn't an 'edge case' - the effective hit window for parts of a circle is still reduced on much more reasonable slider velocities. The example was just the most extreme case I could think of.

Again, even on <5* maps, slider heads can be inconsistent to hit on the side facing away from the slider body, if the slider velocity is anything above ~0.5.

This was posted in the 'ideas' section, because I believe that the classic mod' functionality ( or something similar ) should be default.

**Reply by peppy** [reply]
>  the effective hit window for parts of a circle is still reduced

it's not reduced; it's just moving. and if you read the linked page you will understand that we are aware of this.

**Reply by tovc** [reply]
I'm well aware that it's not actually reduced; thats why I said it's 'effectively reduced'

**Comment by ghost** [comment]
peppy, I am dumbfounded that you would find such a basic gameplay element to be an "edge case scenario". Tech maps are a huge part of the game. Do you expect new players trying to learn tech to figure out which maps need classic mod in order to be playable and to switch back and forth?

**Reply by peppy** [reply, marked as answer]
I highly recommend you follow along and [listen or read through](https://osu.ppy.sh/wiki/en/Community/osu%21_community_meetings) the community meeting notes, as well as the [wiki page](https://github.com/ppy/osu/wiki/Gameplay-differences-from-osu!stable#slider-head-circle-follows-track-snaking-when-not-hit) I linked above. We've already discussed this in great depth, including solutions which address "tech maps".

**Reply by plaguemask** [reply]
The most relevant community meeting excerpts (mainly [this meeting](https://docs.google.com/document/d/13cMCrQN4vvaQFA59BmFHk6D7PH9fwT2ANmhkm5cmzxE)) are talking within the context of slider head _accuracy_, not slider head movement.

Slider head movement is arguably a more significant change. For the entirety of the slider's approach, the target is telegraphed to be in a single spot. But if the player clicks even a fraction of the hit window late, the target has moved from — or possibly completely left — that spot. I'd argue this behavior is counter-intuitive. 

For reference, take the top difficulty of the latest ranked FA map as of writing this comment ([here](https://osu.ppy.sh/beatmapsets/1889094#osu/3890660)). For most sliders in this map, the slider head will have moved completely out of its original spot after 80ms — a span of time within which one cannot feasibly react. An attempt to adjust for this is therefore a crapshoot. Even for time-accurate players, what could have been a near-miss, if hit slightly late, may become _a_ miss if the player's cursor was on the wrong edge of a slider head before it imperceptibly moved.

The linked-above meeting suggests adopting a broad attitude of "beatmaps that heavily rely on stable’s mechanics should not hold back forward progress". I agree to an extent, but while slider head accuracy may be forward progress, I believe slider head movement is a step back given the current state of the game. (Also, in the case where slider head movement is removed, I'd suggest simply scrapping snaking-out sliders. They might look a bit cooler but figuring out how to jury-rig them into a system without slider head movement would be a waste of time.)

**Comment by WitherFlower** [comment]
I've seen quite a lot of people reporting this behaviour as a bug, and it feels to me like the weirdest change to gameplay. In my opinion, such a change should be reconsidered / more feedback should be sought from the community to determine of this is a "good" direction.

**Reply by peppy** [reply]
Please see my other replies above.

**Reply by tovc** [reply]
I never said this was a 'bug'; I only said that one aspect of a good change, negatively affects gameplay on _some_ ( but enough ) maps to justify changing, in my view.

**Reply by ghost** [reply]
The community can and will easily run this game into the ground. I honestly think what is needed at this point is to hire a neutral third party or several with experience in other ranked games.

**Comment by monochrome22** [comment]
The problem:
"Snaking out sliders" follows the sliderball. 
![snakingout](https://user-images.githubusercontent.com/122969285/234853646-f42a57f9-923d-40b5-aede-6bc395d5aac0.png)

The snaking out wouldn't work if the player hits the circle late.
![hittinglate](https://user-images.githubusercontent.com/122969285/234853665-893793bc-33c7-49b4-952b-f4c13da04cc9.png)

How would the snaking out slider catch up to the sliderball?


Possible Solutions:


**Make the beginning of the slider speed up to catch up to the sliderball.** 
This solution is not good, because it interferes with the players ability to determine the speed of a slider. It would also not look good visually, but slider behavior would match stable if snaking out sliders is disabled. 

**Make the sliderball only start moving after the slider head is hit.**
This would cause sliders to be out of rhythm. Currently, you can rely on sliders to get yourself into the correct rhythm.

**Make the sliderball only start moving after the slider head is hit and increase the sliderball travel speed so that it would end at the correct timing.**
This would be a pretty big gameplay change. It would mean sliderballs won't move at a consistent speed every playthrough. Also, if hitting late causes the sliderball to move faster, hitting early would cause it to move slower, allowing players to cheese maps more, at the cost of accuracy. Changing the speed of the slider would also undermine the mappers intentions of how the map should play.

**Make the beginning of the slider teleport to the sliderball.** 
This would not look good visually. However, players already snap to the sliderball when they hit a slider too late, so having the snaking out also snap might not be unintuitive behavior. Also, the gameplay would match stable if snaking out sliders is disabled. In my opinion, this is the best solution, though I don't know exactly how it would look like, maybe it would look worse than I realize.

**Have the snaking out follow the sliderball and have the sliderhead be detached from the slider.**
No.

**Reply by tovc** [reply]
one option would be to have a low opacity 'ghost' slider ball that renders, while keeping the slider head visible until it's either missed or hit, then increasing the opacity of said ghost slider ball to a normal level

this still doesn't adress the issues with snaking sliders

**Reply by monochrome22** [reply]
As you said, your idea wouldn't address the issue with snaking sliders.

The idea of using opacity is interesting though. Maybe the part of the slider before the sliderball could progressively decrease in opacity, and by the time the slider head misses and disappears, it will be completely faded. The transparent part would have to immediately disappear when the slider head is hit or always have the transparent part be visible, even if the slider was hit on time. I'm not sure if this would be better than teleporting. Would be great if someone could make proper demos. 

But it would look something like this:
![opacity2](https://user-images.githubusercontent.com/122969285/234911816-d1476b62-f926-4604-87ef-c962f8805d6d.png)



Another idea would be to have the opacity of the entire slider before the sliderball progressively decreasing until the sliderball reaches the end. I wonder what this would look like. It would probably defeat the purpose of snaking sliders though.

**Reply by tovc** [reply]
snaking sliders are purely a visual thing; we've done without them for over a decade in stable. Gameplay should come before visual flair. Making the track before the slider ball fade is probably one of the better comprimises here

**Reply by monochrome22** [reply]
I managed to make some simple previews of what fading would look like. This is made so the fading ends by the end of the hit window.
In case it's not clear, there is supposed to be a slider head at the start which misses by the time fading finishes, I wasn't able to put it on my preview.


Short duration slider (around 150ms): 
![Short Duration](https://user-images.githubusercontent.com/122969285/235159440-5a408d29-75a4-4e29-8b50-138e589aa457.gif)

Average duration slider (around 250ms): 
![Average Duration](https://user-images.githubusercontent.com/122969285/235159494-4953f498-cc0a-44f1-af29-c70e319621a2.gif)

Long duration slider (around 400ms): 
![Long Duration](https://user-images.githubusercontent.com/122969285/235159531-c0fe4216-6901-4e38-98ed-b7cb74910081.gif)

My opinion: 
The feel of transitioning from full slider to partial slider might be slightly awkward. However, if you hit late and need to jump from slider head to sliderball, then it should be comfortable, because you can still see the slider. Fading seems like a good middle ground for having snaking sliders work and having hit windows work. Hard to tell without a proper gameplay test. Also, I have no idea how devs feels about this direction. But having fading on snaking out sliders is probably better than having a moving hit location, which has a significant negative impact on gameplay.

**Reply by tovc** [reply]
How would the slider body between the head and ball fade after hitting it? I feel like an increased fade speed would feel awkward; Having it disappear instantly also seems like it'd be wrong too. I suppose having it simply fade at the same rate and allowing the slider head to mask it would work.

Another idea that I had that would avoid transparency would be to simply extend the length of the slider 'ball' and follow 'circle' down the length of the body while the head isn't being played. I put those terms in quotes because they would be more akin to a slider follow 'sausage' at that stage. I think this would be bordering on too radical for a lot of people.

The irony of this discussion is that a majority of sliders that are affected by this, are repeat sliders anyway, so fading the body isn't even necessary.

That last part is very true; gameplay should always come before visual flair. Snaking sliders are just that, and they introduce this conflicting gameplay element. The Devs themselves aren't the ones that primarily play the game let alone maps that are affected by this. I imagine that most players would tell you that the existing functionality feels wrong and would prefer classic functionality over, again, exclusively visual flair.

**Reply by monochrome22** [reply]
> How would the slider body between the head and ball fade after hitting it? 

Always fading it to late hit window is probably best, even if the slider head is hit or if the slider is too short. 

> Another idea that I had that would avoid transparency would be to simply extend the length of the slider 'ball' and follow 'circle' down the length of the body while the head isn't being played. I put those terms in quotes because they would be more akin to a slider follow 'sausage' at that stage. 

Definitely not.

> The irony of this discussion is that a majority of sliders that are affected by this, are repeat sliders anyway, so fading the body isn't even necessary.

It's true that fading wouldn't be necessary for reverse sliders.

> gameplay should always come before visual flair. Snaking sliders are just that, and they introduce this conflicting gameplay element.

You probably won't get anywhere with arguments from this angle. The developers want lazer to have new features, snaking sliders being one of them. Instead of just disabling them or making them look broken, their approach is to try to figure out how to make them work, which I think is reasonable. Late sliders are an issue not because developers don't care, but because they haven't focused on balance too much yet, it's still a work in progress. In the wiki that peppy linked, late sliders are labeled as "Needs further consideration".

**Reply by tovc** [reply]
In my understanding, peppy's attitude towards lazer is that you'll switch when you switch. Stable will exist for as long as people will play it. The last thing we want is a divided playerbase. I am sure that simply saying 'visuals don't matter' won't get us anywhere, but that's not what I'm saying. That said, osu! only works because people pay for it. A divisive change like moving slider heads is enough reason for many players to stick with the tried and true stable.

**Reply by Sleepteiner** [reply]
> The developers want lazer to have new features, snaking sliders being one of them.

It is baffling to me that a gameplay change that objectively hurts thousands and thousands of ranked maps is being pushed just because "the devs want the game to look different." Any change that hurts the long history of mapping to anywhere near this extent should not even be considered without a very big justification and "the developers want lazer to have new features" is not a justification at all.

**Reply by bdach** [reply]
> objectively hurts thousands and thousands of ranked maps

Citation needed on that one.

Also did you skip over the rest of this discussion, that describes the rationale for doing this and also describes that the old behaviour can still be restored by way of the classic mod?

**Reply by Sleepteiner** [reply]
> Citation needed on that one.

Every map that uses fast sliders.

As mentioned in the original post, this change will create a situation that rewards players for hitting sliders early. In a rhythm game.

> Also did you skip over the rest of this discussion, that describes the rationale for doing this

I just read through this discussion again. Where is the rationale? "Snaking out sliders" is an unnecessary feature. It does not warrant a detrimental gameplay change to make it work.

> the old behaviour can still be restored by way of the classic mod

Classic mod has never been and will never be a solution to stuff like this with how the mod is designed. The vast majority of players will not use the classic mod and the majority won't even know what it does. If a change like this becomes the default nomod behavior on all maps, then the vast majority of plays on the maps affected from that point on will be on a worse version of that map forever. That is sad to me.

**Reply by monochrome22** [reply]
Well I agree it shouldn't be part of classic mod and that sliders should function same way as stable by default, but like I said in my previous post, you will probably not get anywhere by trying to argue that snaking out sliders should just be removed, you're better off spending your time coming up with ideas on how to make them work.

I asked peppy about fading the slider before sliderball, and he said (paraphrasing): "I don't think it will fix the issue". I would like to say that I think there's a decent chance that fading will look better than expected if actually implemented. If no better ideas are presented, then fading should definitely at least be tested to see what it looks like in actual gameplay and how it feels.

Another idea is to have the slider head be detached and connect it to the sliderball with a line. The line would disappear (or fade the same way as the slider head) as soon as the slider head is hit or missed. The line can be same color as combo color. This might also look ugly, but it's just another idea.
![LineConnect](https://github.com/ppy/osu/assets/122969285/f3d8b202-8bc0-44d9-9f0e-c304a9655f6d)

**Reply by Sleepteiner** [reply]
That is a novel idea, but I feel like it would make the game look very cluttered and confusing which would make it harder to read slider heavy maps and ruin the visual design of many maps. Snaking out sliders in general ruins the visual design of many maps because mappers design many kinds of patterns with sliders staying on screen in mind.

**Comment by Zyfarok** [comment]
Let me add a bit on this subject :
The hit-window at OD 8 is around ~120ms, while the reaction time of a human is about ~200ms (see: https://en.wikipedia.org/wiki/Mental_chronometry and https://humanbenchmark.com/tests/reactiontime for example), thus if you miss-timed the slider, you won't even have time to realize that the head started moving when you press your key to hit it (unless you are so late that you are out of the hit-window to begin with). Thus, there's is no way you could have aimed at the moved slider-head since you (probably) didn't yet realize you were late and also your brain didn't have enough time to process any hint on the slider velocity yet. (You didn't see the slider-ball moving yet)

Concerning the visual aspects, if you are late to hit, then it's fine to have the slider-head teleporting to catch-up once you hit the head. The ball could have started moving ahead on it's own, but the head shouldn't move until you hit it ! If the hit-window was much larger, and the slider was moving slowly it could make sense but with the hitwindows of osu! and the fast moving sliders of high-SR beatmaps it really doesn't. The visual aspect should not prevail over the playability, and there are multiple alternative to make it look acceptable even if it is not as intuitive to look at when watching a replay. Any solution that allows you to visualize the actual slider velocity and timing as soon as possible while not impacting the aiming of the head should work.

One alternative could be to accept any hit that is either on the (non-moving) slider-head, or on the already moving slider-ball or anywhere in-between on the slider-track. This would avoid punishing players that are late on the tapping but not on the aim, even though this is contradictory with the normal way of playing the game which requires synchronizing tapping and aim, such that you tap when you are inside the circle and preferably close to the center and also slowing down your cursor as you approach the circle to reduce the risk of missing.

**Reply by tovc** [reply]
Reaction time is not the same as a hit window. You don't hit objects on reaction, you time them with the music and with the approach circle, if applicable. You're definitely capable of hitting a slider late, if the slider is a reasonable speed, and you know that you're aiming for it. This discussion concerns the 'feel' of hitting where a slider head _was_. It has nothing to do with reaction time.

While I do also feel quite strongly about this issue, I don't think that misrepresenting the issue is valuable.

**Reply by Zyfarok** [reply]
> Reaction time is not the same as a hit window.

I know, but you are completely missing my point.

> You don't hit objects on reaction, you time them with the music and with the approach circle, if applicable.

You act based on the visual and audio clues that have been given to you. In particular, you aim according to the position you have seen the circle to be, and the reaction time (or "brain lag") defines the minimum amount of time between the time at which the photons of an image reaches your eye and the time at which your hand movements can potentially be affected by what you saw in that image. If you saw the circle moving, you would potentially predict the movement to compensate your reaction time and could aim correctly. My point is: here, EVEN IF you are hitting at the latest acceptable for a 50 (120ms late at OD 8), then 1 brain-lag before (150~200ms at least) the circle was not moving yet and thus it was impossible for you to predict the movement of the slider-head, unless you already knew the slider-velocity of that slider as well as how late you would be hitting !

I don't think it is any reasonable to expect the players to predict and account for the slider-velocity (which can vary a lot from sliders to sliders on some maps) and the timing of their hit in how they aim slider.

The 'feel' you talk about is at least partially explained by the reasoning above, and I believe providing this explanation is valuable as it helps in showing that this is not just subjective reasons like "being used to how sliders worked before" but just because this new game-play mechanic objectively doesn't make sense to begin with.

**Reply by tovc** [reply]
Again, you do not 'react' to slider head movement. Sliders begin moving when the approach circle matches the slider head circle's size. At AR9.5, that gives you 525ms to react to a slider's head movement.

Sure, you can't immediately tell how _fast_ a slider's movement is, but that is largely irrelevant due to the follow window allowing lenience, and slider speeds being largely simple enough on ranked maps. In addition to that, only slider follow points and tails need to be hit. If we're being anal here, you're prefectly capable of simply moving your cursor over the follow points/end points to avoid breaking combo. Beyond actually reacting to a slider's speed, ranked maps do a good job of introducing slider speeds to you, for the most part. More often than not, slider breaks ( aside from those caused by misaiming ) are caused by sliders being unexpectedly slow, rather than unexpectedly fast.

That entire conversation is superfluous anyway, as we're not talking about slider ball speeds, we're talking about slider head hit windows.

**Reply by Non-Binary-Programmer** [reply]
> Reaction time is not the same as a hit window. You don't hit objects on reaction, you time them with the music and with the approach circle, if applicable. You're definitely capable of hitting a slider late, if the slider is a reasonable speed, and you know that you're aiming for it.

I don't disagree with this statement. However, the qualifier "if... you know that you're aiming for it," will almost never be fulfilled in normal gameplay. Because you don't hit objects on reaction, you try to hit objects on time every time, unless you are intentionally cheesing a pattern or have run out of stamina. This means that, regardless of your actual timing, and therefore where you would actually need to hit the sliderhead, you try to hit the sliderhead as if you had perfect timing, meaning that you miss it if you are actually late.

**Comment by ebolite** [comment]
I feel I should add to this as a relatively good player who mainly plays lazer. This is an issue I feel very strongly about.

**Slider head movement doesn't work well on ANY map above 6 stars.** This is not a change the community or mappers can adapt to. I firmly believe that slider head movement is the thing that makes lazer feel "different" to many players who can't quite put it into words. You wouldn't think it would be very noticeable, but it is actually extremely significant in practice. I can usually notice pretty quickly if I forgot to put on my no-SHM preset.

https://github.com/ppy/osu/assets/22662047/8ddfbc39-7aa7-4973-82c2-1f7f4c160dae

This slider isn't a high slider velocity at all, yet I still miss in (what I assume) is the 300 hit window due to tapping on the edge of where the hitcircle used to be. You can see I immediately quit to go put on my preset because I know exactly why I missed. The reason that there isn't much community backlash on this yet is because hardly anyone knows this is a thing. If this mechanic is left in as-is, there will be a lot of complaints from good players who decide to switch to lazer.

**Comment by monochrome22** [comment]
There was a [discussion in Discord](https://discord.com/channels/188630481301012481/1097318920991559880/1128796372833927218) regarding solutions for the slider head movement. 
Wieku provided previews while others provided feedback. 
This post is a summary of the discussion.

---

Preview for sliderhead speeding up to catch up to the sliderball:

https://github.com/ppy/osu/assets/122969285/590a9360-dced-4f81-a8e0-643f3de36145

The consensus was that this looks jarring.

---

Preview for beginning of the slider teleporting to the sliderball:

https://github.com/ppy/osu/assets/122969285/28c2c629-e36b-408b-8874-bbab31365dce

This also looks jarring.

---

Preview for disconnected sliderheads:

https://github.com/ppy/osu/assets/122969285/d237445c-c2c6-4e9c-bf9c-db7cc33f52fe

This looks surprisingly good. However, the disconnected sliderheads can look out of place. This is particularly visible at 0:38 of the preview, where it feels like there are sliderheads in the middle of nowhere.

---

Preview for the part of the slider between the sliderhead and sliderball progressively fading from 100% opacity to 0% opacity:

https://github.com/ppy/osu/assets/122969285/2a9c6c47-7c9e-45cf-a9fb-1b6fc2b5ec1d

---

Preview for progressively fading on OD 0: 

https://github.com/ppy/osu/assets/122969285/7a5afbab-2156-490e-9992-37f4d13848a8

---

Preview for fading always being at 30% opacity until sliderhead is missed on OD 0:

https://github.com/ppy/osu/assets/122969285/183fed1a-7270-4206-aa6d-529356f47096

---

Then there was talk about whether fading should always appear, even if sliderhead is hit on time, or if it should appear only if the sliderhead is hit late. We agreed that fading should only appear if the sliderhead is hit late, and the fading part of the slider should quickly fade out once the sliderhead is hit. 

Preview for 30% opacity with fading only appearing late and disappearing quickly once the sliderhead is hit:

https://github.com/ppy/osu/assets/122969285/98919091-2867-46d0-8181-41237066e8ba

In this preview, note the UR bar. Every consecutive hit is done later than the previous one.
The slider(7) at 1:02 is a good example of what fading would look like in normal cases where the slider is hit late. Also (3) at 1:04.

---

After the previews, peppy said that any further previews should be done in the lazer client, as it allows for better detection of edge cases.

There wasn't an agreement on exactly how the fading should work. 30% might be too low. If the opacity is too low, then sliderheads will look disconnected from the slider, which looks visually broken. If the opacity is too high, then the fading will undermine the snaking out slider function, because it will look like it's not even snaking.
To find the exact percentages, more precise previews need to be done in the lazer client. The goal is to find the lowest value where sliderheads don't look disconnected. And it's not clear whether there should be some progressive fading (for example, from 60% to 30%) or if it should be constant.
Currently we are waiting on someone to make more precise previews in the lazer client.

---

Some suggestions that were put forward:
**With fading, make the sliderbody fade by the time the sliderend is reached, rather than by the end of the hit window**
This caused sliderheads to be disconnected from the sliderbody, which looked broken.

**Use 50% opacity for fading**
This undermined the snaking out function too much. Some sliders looked like they weren't even snaking.

**For teleporting, make the sliderball appear at the right time instead of when the sliderhead is missed**
This likely wouldn't make much of a difference, it would probably still look jarring.

---

Personal thoughts:
Out of the previews, fading is the only one that doesn't look visually broken.
I think progressive fading is better than constant, because it creates the sense of timing window closing, mimics the snaking out function by disappearing over time, and is good for making sure sliderheads don't look disconnected. One concern I have with constant fading is that even at something like 40% opacity, if someone is playing with low brightness or with dark slider borders, then they might not even see the fading. Progressive fading should guarantee that nothing ever looks visually broken, at the cost of undermining snaking sliders. Though with fading only appearing if the sliderhead is hit late, fading undermining snaking sliders might not even be an issue.

**Comment by Non-Binary-Programmer** [comment]
As someone who has been a lazer main for over 2 years now, I would like to say that this is one of my two biggest complaints with lazer (the other being website display of pp being strictly lower on maps with sliders, but that one is being worked on). I will give the disclaimer that none of this is based on technical analysis, and is rather an example of the experience of a relatively average player. Even on relatively simple jump maps with low slider velocities, it feels very significantly worse to try to hit sliders than hit circles. For this reason, I have always had a preference for maps with few to no sliders in high difficulty sections, although I never really knew why before I read about this.

I have a few problems with the approach of making this the default behavior and having an option in classic mod to revert this change:

1. It's a widespread problem.
In my experience playing the game, basically every map with sliders has this problem. Although it would theoretically be less extreme in easier maps with lower slider velocities, I have not found this to be the case, perhaps because the longer hit windows exacerbate the difference between the actual and original sliderhead location (although this is conjecture).

2. I didn't know it was an option.
Before reading this post, I had no way of even knowing that this was the behavior of sliderheads, because I already disabled snaking sliders for performance reasons and was used to how sliders worked in stable. I just assumed that, because I avoided playing maps with sliders, I was misreading them, causing me to think I had aimed them properly when I actually hadn't.

3. It's an unintuitive mechanic.
The idea that, in order to hit a sliderhead properly regardless of whether you time it properly, you need to move your cursor away from where the approach circle meets the sliderhead, is very unintuitive, at least for me. This is because with the main other hit object, hit circles, the opposite is actually true: you want to have your cursor be where the approach circle meets the hit circle when you tap/click it, regardless of when you time your tap/click.

In the end, I want to reiterate that this is just my experience. I understand the reasons for the current default behavior, even if I disagree with them. Ultimately, I can't quantitatively express my discomfort with slider head movement (although others have), so I'm mainly writing this to make clear to the developers that average players playing average maps can feel these changes, and they're not specific to "edge case maps" like high sv slidertech (although I personally believe that these maps, because they are rankable, should not have their behavior considered to be an edge case in the same way that aspire maps are).

**Comment by peppy** [comment]
This has been resolved via https://github.com/ppy/osu/pull/24810!
